README.md

```diff
@@ -21,49 +50,120 @@ through ssh.
 a toolchain or two...
 
 ```
+# check available cross-compilation targets
+rustup target list
+
 # a few ios devices
 rustup target install aarch64-apple-ios
 rustup target install armv7-apple-ios
 rustup target install arm-linux-androideabi
-
 # a suitable ios simulator
 rustup target install x86_64-apple-ios
 
-# android arm devices
+# some android devices
 rustup target install arm-linux-androideabi
+rustup target install armv7-linux-androideabi
+rustup target install aarch64-linux-android
+rustup target install i686-linux-android
+rustup target install x86_64-linux-android
 
-# for a Raspberry Pi 2
+# a Raspberry Pi
 rustup target install armv7-unknown-linux-gnueabihf
 ```
 
-Let's install dinghy...
+We will need some configuration bits here. So let's create a `.dinghy/dinghy.toml` in
+your home directory to specify available platforms with their rustc triple:
+```toml
+# For ios
+[platforms.ios-armv7]
+rustc_triple="armv7-apple-ios"
+
+[platforms.ios-armv7s]
+rustc_triple="armv7s-apple-ios"
 
-```
-cargo install dinghy
+[platforms.ios-arm64]
+rustc_triple="aarch64-apple-ios"
+
+[platforms.ios-i386]
+rustc_triple="i386-apple-ios"
+
+[platforms.ios-x86_64]
+rustc_triple="x86_64-apple-ios"
+
+
+# For android
+[platforms.android-arm]
+rustc_triple="arm-linux-androideabi"
+toolchain="<path to your arm-linux-androideabi standalone toolchain>"
+
+[platforms.android-armv7]
+rustc_triple="armv7-linux-androideabi"
+toolchain="<path to your armv7-linux-androideabi standalone toolchain>"
+
+[platforms.android-arm64]
+rustc_triple="aarch64-linux-android"
+toolchain="<path to your aarch64-linux-android standalone toolchain>"
+
+[platforms.android-x86]
+rustc_triple="i686-linux-android"
+toolchain="<path to your i686-linux-android standalone toolchain>"
+
+[platforms.android-x86-64]
+rustc_triple="x86_64-linux-android"
+toolchain="<path to your x86_64-linux-android standalone toolchain>"
+
+
+# A raspbian platform
+[platforms.raspbian]
+rustc_triple="arm-unknown-linux-gnueabihf"
+toolchain="<path to your raspbian toolchain>"
 ```
 
 And... stop here. Unfortunately, there is a bit of stuff to do by hand before
 the fun stuff come.
 
 ## Android setup
 
-You'll need the usual ANDROID_NDK_HOME, and `adb` somewhere in your path.
-`fb-adb` will be used if its available on the path, giving better error
-handling.
+You'll need to make a stand-alone toolchain as explained in the [official Android documentation](https://developer.android.com/ndk/guides/standalone_toolchain.html) (Dinghy cannot use the NDK directly), and `adb` somewhere in your path.
+`fb-adb` will be used if its available on the path. It allows dinghy to forward correctly binaries/benches/tests exit codes from the conneced device to the host platform.
 Also, your phone must have developer options enabled.
 
-## iOS simulator setup
+You will also probably have to set up a toolchain: rustc needs a working linker
+to be able to generate executable. On my case, for using on a RaspberryPi, I
```

Comment:
raspberry pi in section android setup  ??

Reply:
k